lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -507,6 +508,25 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
   return success();
 }
 
+LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
+                                     memref::CopyOp copyOp) const {
+  auto source = copyOp.getSource();
+  auto target = copyOp.getTarget();
+
+  auto srcType = source.getType().cast<MemRefType>();
+  auto tgtType = target.getType().cast<MemRefType>();
+
+  assert(srcType.getElementType() == tgtType.getElementType() &&
```

Comment:
I think these invariants should be enforced upstream by the `memref::CopyOp` verifier, I don't think they need to be asserted here.